lib/elixir/lib/list.ex

```diff
@@ -275,6 +275,26 @@ defmodule List do
     :lists.keydelete(key, position + 1, list)
   end
 
+
+  @doc """
+
+  ## Examples
+
+      iex> List.keytake([a: 1, b: 2], :a, 0)
+      {:value, {:a, 1}, [b: 2]}
+
+      iex> List.keytake([a: 1, b: 2], 2, 1)
+      {:value, {:b, 2}, [a: 1]}
+
+      iex> List.keytake([a: 1, b: 2], :c, 0)
+      false
+
+  """
+  @spec keytake([tuple], any, non_neg_integer) :: {:value, tuple, [tuple]} | false
```

Comment:
I think we can improve the function return, make it either `{any, [tuple]}` or `{tuple, [tuple]} | nil`. @josevalim what are your thoughts?